generate-examples/generate-cert-chain.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+# This script generates a 3-cert ECDSA chain (root -> CA -> issuer) valid for 10 years.
+# Leaf cert uses P-256 (as per the SMART Health Card Framework), CA and root CA use the
+# increasingly stronger P-384 and P-521, respectively (simulatate real PKI).
+
+# directory where intermediate files are kept
+tmpdir=certs
+mkdir -p $tmpdir
+
+# generate self-signed root CA cert
+openssl req -x509 -new -newkey ec:<(openssl ecparam -name secp521r1) -keyout $tmpdir/root_CA.key -out $tmpdir/root_CA.crt -nodes -subj "/CN=SMART Health Card Example Root CA" -days 3650 -config openssl_ca.cnf -extensions v3_ca
+
+# generate intermediate CA cert request
+openssl req -new -newkey ec:<(openssl ecparam -name secp384r1) -keyout $tmpdir/CA.key -out $tmpdir/CA.csr -nodes -subj "/CN=SMART Health Card Example CA" -config openssl_ca.cnf -extensions v3_ca
+
+# root CA signs the CA cert request
+openssl x509 -req -in $tmpdir/CA.csr -out $tmpdir/CA.crt -CA $tmpdir/root_CA.crt -CAkey $tmpdir/root_CA.key -CAcreateserial -days 3650 -extfile openssl_ca.cnf -extensions v3_ca
+
+# generate issuer signing cert request
+openssl req -new -newkey ec:<(openssl ecparam -name prime256v1) -keyout $tmpdir/issuer.key -out $tmpdir/issuer.csr -nodes -subj "/CN=SMART Health Card Example Issuer" -config openssl_ca.cnf -extensions v3_req
+
+# intermediate CA signs the issuer cert request
+openssl x509 -req -in $tmpdir/issuer.csr -out $tmpdir/issuer.crt -CA $tmpdir/CA.crt -CAkey $tmpdir/CA.key -CAcreateserial -days 3650 -extfile openssl_ca.cnf -extensions v3_req
```

Comment:
I would suggest a shorter lifetime for the sample end entity cert to be more like a production PKI, maybe 3 years?

Reply:
I was debating that, and wanted to keep the chain (and the spec examples) valid for a long time. Thinking about this more, I guess having a end-user cert of 1 year would match better the recommendation of the spec, and be a better "example". We could either re-gen examples every year, or have expired certs (but valid cards, signed during the validity period) in the spec. I tend to lean towards shorter certs now...

Reply:
Agreed, given that we make recommendations in the spec it seems fair to align to them

Reply:
I'll use 1 / 5 / 10 years, matching the spec for issuer cert, and real-life-ish PKI deployment for CAs.